src/Analyzers/CSharp/Analyzers/SimplifyLinqExpression/CSharpSimplifyLinqExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,179 @@
+                   title: new LocalizableResourceString(
+                       nameOfLocalizableResource: nameof(CSharpAnalyzersResources.Simplify_Linq_expression),
+                       resourceManager: CSharpAnalyzersResources.ResourceManager,
+                       resourceSource: typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var whereMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+            var linqMethods = ImmutableHashSet.CreateBuilder<IMethodSymbol>();
+
+            var fullyQualifiedEnumerableName = typeof(Enumerable)?.FullName;
+            if (fullyQualifiedEnumerableName is null)
+            {
+                return;
+            }
+
+            var enumerableNamedType = context.Compilation.GetTypeByMetadataName(fullyQualifiedEnumerableName);
+            if (enumerableNamedType is null)
+            {
+                return;
+            }
+
+            var enumerableMethods = enumerableNamedType.GetMembers(nameof(Enumerable.Where)).OfType<IMethodSymbol>();
+            whereMethods.UnionWith(enumerableMethods);
+
+            // add all valid linq calls
+            foreach (var id in _nonEnumerableReturningLinqMethodNames)
+            {
+                enumerableMethods = enumerableNamedType.GetMembers(id).OfType<IMethodSymbol>();
+                linqMethods.UnionWith(enumerableMethods);
+            }
+
+            if (whereMethods.Count > 0 && linqMethods.Count > 0)
+            {
+                context.RegisterOperationAction(
+                    context => AnalyzeAction(context, whereMethods.ToImmutable(),
+                    linqMethods.ToImmutable()),
+                    OperationKind.Invocation);
+            }
+        }
+
+        public void AnalyzeAction(OperationAnalysisContext context, ImmutableHashSet<IMethodSymbol> whereMethods, ImmutableHashSet<IMethodSymbol> linqMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var semanticModel = context.Operation.SemanticModel;
+            var invocationArgument = invocationOperation.Arguments.FirstOrDefault();
+
+            // Check to make sure the invocation argument is an InvocationExpressionSyntax
+            // Example: If invocationOperation syntax is Data().Where(...).Single(), then invocationArgument is Data.Where(...)
+            if (invocationArgument is null)
+            {
+                return;
+            }
+
+            var targetOperation = invocationArgument.Children.FirstOrDefault(c => c is IOperation);
```

Comment:
i don't understand what this means.  all Children will be IOperations.  So this `is` doesn't seems necessary.

Reply:
1. Why is this using .Children instead of the strongly typed `.Instance`
2. why not call tihs `Instance`?